packages/react/src/utilities/decorators/withViewport.tsx

```diff
@@ -102,7 +102,9 @@ export function withViewport<TProps extends { viewport?: IViewport }, TState>(
           this._events.on(win, 'resize', this._onAsyncResize);
         }
 
-        this._updateViewport();
+        setTimeout(() => {
```

Comment:
the timeout is not cleared on unmount, so there is potential that _updateViewport is called on an unmounted component.

Reply:
doh

Reply:
I ended up using the Async class approach to align with the other class components.